pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -157,13 +155,15 @@ func (gc *GarbageCollector) Run(ctx context.Context, workers int) {
 
 	go gc.dependencyGraphBuilder.Run(ctx)
 
-	if !cache.WaitForNamedCacheSync("garbage collector", ctx.Done(), func() bool {
+	if !cache.WaitForNamedCacheSync("garbage collector", waitForStopOrTimeout(ctx.Done(), 30*time.Second), func() bool {
 		return gc.dependencyGraphBuilder.IsSynced(logger)
```

Comment:
if there are unsyncable resources present at startup, won't this still hit the 30 second timeout, WaitForNamedCacheSync will return false, and `return gc.dependencyGraphBuilder.IsSynced(logger)` will return false and never start GC?

seems like we need a test for that

Reply:
Hi @liggitt, I'm following up this patch in #125796. According to the patch(next a few lines after your comment), the gc gets started no matter IsSynced() returns true or false. It just affects the output log. 
In fact, I doubt if this `WaitForNamedCacheSync` should be kept here. Since without this,  we're still able to output err logs in https://github.com/kubernetes/kubernetes/blob/fa75b4371d481650d935bf09da934cd9222c1cea/cmd/kube-controller-manager/app/core.go#L705
https://github.com/kubernetes/kubernetes/blob/fa75b4371d481650d935bf09da934cd9222c1cea/pkg/controller/garbagecollector/garbagecollector.go#L268

Reply:
you're right, I misread the IsSynced return and missed it was inside a closure